### PR TITLE
Stop storagenode on ctrl+c

### PR DESCRIPTION
**Pull request by mniewrzal**

### CODE REVIEW

File: cmd/storagenode/main.go [hunk 1]
```diff
@@ -169,8 +171,15 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
...
+	var closeError error
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-c
+		closeError = peer.Close()
+	}()
+
 	runError := peer.Run(ctx)
-	closeError := peer.Close()
```
Comment (egonelbre) [inline cmd/storagenode/main.go]: This fails to close peer when there's no signal.

Comment (mniewrzal) [reply cmd/storagenode/main.go]: ok, maybe one of the pieces is not canceling itself. I will take a look.

File: pkg/server/server.go [hunk 2]
```diff
@@ -82,5 +82,13 @@ func (p *Server) Run(ctx context.Context) (err error) {
 		return next.Run(ctx, p)
 	}
 
+	go func() {
+		select {
+		case <-ctx.Done():
+			// error is always nil
+			_ = p.Close()
```
Comment (egonelbre) [inline pkg/server/server.go]: Use errgroup, it's not necessarily nil in the future:

```
ctx, cancel := context.WithCancel(ctx)
defer cancel()

var group errgroup.Group
group.Go(func() error {
    <-ctx.Done()
    return p.Close()
})
group.Go(func() error {
    err := p.grpc.Serve(p.lis)
    cancel()
    return err
})

return group.Wait()
```

File: scripts/check-travis-tidy.sh [hunk 3]
```diff
@@ -1,12 +1,12 @@
 #!/bin/bash
 set -ueo pipefail
```
Comment (egonelbre) [inline scripts/check-travis-tidy.sh]: this needs to be removed as well

**Pull request merged**